src/vec3.h

```diff
@@ -1,35 +1,38 @@
 #include <cmath>
-#include <stdlib.h>
+#include <array>
+#include <iostream>
+
 
 class Vec3
 {
+
 private:
-    float v[3];
+    std::array<float, 3> v;

```

Comment:
Unica modifica di struttura al tuo Vec3.
Se non ti piace, torniamo a `float[3]`, no problem.

Reply:
@Ale32 
In pratica la differenza tra `float[3]` e `std::array<float, 3>` sta nella gestione a livello di implementazione, e nel fatto che gli array come `float[]` sono considerati C-like, mentre `std::array` e' considerato C++-like.

- stackoverflow: [Replace fixed size arrays with std::array?](https://stackoverflow.com/a/22571173/1162467)
- quora: [Why is there std: :array<> in C++11? What are the benefits of it?](https://www.quora.com/Why-is-there-std-array-in-C++11-What-are-the-benefits-of-it)
- Migrating from C to C++: [An Introduction to std::array](https://embeddedartistry.com/blog/2017/6/28/an-introduction-to-stdarray)

Da cppreference.com, [`std::array`](http://en.cppreference.com/w/cpp/container/array) ha tutta una serie di funzionalita' in piu' che i built-in array non ti consentono di avere, e che fa comodo quando lavori con altre funzioni della standard library C++.

Una tra tutte, la presenza di costruttore e distruttore, che si occupano per te di liberare la memoria dell'array e di deallocare i puntatori che puoi aver usato per gestirne gli accessi.

Ha i metodi [`fill`](http://en.cppreference.com/w/cpp/container/array/fill) e [`swap`](http://en.cppreference.com/w/cpp/container/array/swap) che so sempre comodi, e ha tutta una serie di operatori pronti all'uso, ad esempio per fare comparazione.

Nella documentazione su cppreference.com, [nella parte degli esempi](http://en.cppreference.com/w/cpp/container/array#Example), puoi farti un idea dei vantaggi che offre rispetto a un array built-in.